addons/controllers/packageinstallstatus_controller.go

```diff
@@ -0,0 +1,434 @@
+	// If it does, just update it with the new condition
+	var conditionExists bool
+	var conditionChanged bool
+	for i, existingCond := range clusterBootstrapStatus.Conditions {
+		if !strings.Contains(string(existingCond.Type), strings.Title(pkgShortname)) {
+			continue
+		}
+		conditionExists = true
+		if !util.HasSameState(&clusterBootstrapStatus.Conditions[i], &condition) {
+			conditionChanged = true
+			clusterBootstrapStatus.Conditions[i] = condition
+		}
+	}
+	if !conditionExists {
+		conditionChanged = true
+		clusterBootstrapStatus.Conditions = append(clusterBootstrapStatus.Conditions, condition)
+	}
+
+	return conditionChanged, nil
+}
+
+// watchPackageInstalls sets a remote watch on the provided cluster on the Kind resource
+func (r *PackageInstallStatusReconciler) watchPackageInstalls(ctx context.Context, cluster *clusterapiv1beta1.Cluster, log logr.Logger) error {
+	// If there is no tracker, don't watch remote package installs
+	if r.Tracker == nil {
+		return nil
+	}
+
+	return r.Tracker.Watch(ctx, &remote.WatchInput{
+		Name:         "watchPackageInstallStatus",
+		Cluster:      clusterapiutil.ObjectKey(cluster),
+		Watcher:      r.controller,
+		Kind:         &kapppkgiv1alpha1.PackageInstall{},
+		EventHandler: handler.EnqueueRequestsFromMapFunc(r.pkgiToCluster),
+		// ClusterBootstrap resource only exists in the management cluster, hence using local client
+		Predicates: []predicate.Predicate{r.PkgiStatusChanged(log), predicates.ClusterHasLabel(constants.TKRLabelClassyClusters, r.Log)},
+	})
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *PackageInstallStatusReconciler) SetupWithManager(ctx context.Context, mgr ctrl.Manager, options controller.Options) error {
+	pkgiStatusController, err := ctrl.NewControllerManagedBy(mgr).
+		For(&clusterapiv1beta1.Cluster{}).
+		Watches(
+			&source.Kind{Type: &kapppkgiv1alpha1.PackageInstall{}},
+			handler.EnqueueRequestsFromMapFunc(r.pkgiToCluster),
+			builder.WithPredicates(r.PkgiStatusChanged(r.Log)),
+		).
+		WithOptions(options).
+		WithEventFilter(predicates.ClusterHasLabel(constants.TKRLabelClassyClusters, r.Log)).
+		Build(r)
+
+	if err != nil {
+		return errors.Wrap(err, "failed setting up with a controller manager")
+	}
+
+	r.controller = pkgiStatusController
+	r.Ctx = ctx
+	r.liveClient, err = client.New(mgr.GetConfig(), client.Options{Scheme: mgr.GetScheme()})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// pkgiToCluster returns a list of Requests with Cluster ObjectKey
+func (r *PackageInstallStatusReconciler) pkgiToCluster(o client.Object) []ctrl.Request {
+	pkgi, ok := o.(*kapppkgiv1alpha1.PackageInstall)
+	if !ok {
+		r.Log.Error(errors.New("invalid type"),
+			"Expected to receive PackageInstall resource",
+			"actualType", fmt.Sprintf("%T", o))
+		return nil
+	}
+
+	log := r.Log.WithValues("pkgi-name", pkgi.Name)
+	log.V(4).Info("Mapping PackageInstalls to cluster")
```

Comment:
suggestion: usually we want log level to be configurable, maybe lets use log.Info() for now and enhance it later.